### ISSUE TITLE
[Discussion] Reorganize bedrock documentation

### ISSUE THREAD

**Issue by maureenlholland** [mozilla/bedrock#11682]
We could use a high level reorganization of the bedrock documentation to make the Mozorg and Pocket mode behaviour really clear. 

It would also be a nice opportunity to breakdown some of the larger sections such as [Installing bedrock](https://bedrock.readthedocs.io/en/latest/install.html) and [Developing on bedrock](https://bedrock.readthedocs.io/en/latest/coding.html)  

Splitting this task out from https://github.com/mozilla/bedrock/issues/11390 for easier-to-track discussion.
Breakdown of top level pages by wordcount and heading level here: 
[bedrock docs - pages.pdf](https://github.com/mozilla/bedrock/files/8763891/bedrock.docs.-.pages.pdf)

**Comment by reemhamz** [comment]
What are some ways we could reorganize? Interested to hear your thoughts, I'd love to make the documentation cleaner :)

**Comment by maureenlholland** [comment]
@reemhamz mostly I'd like to add subheadings in the side nav (like they do for readthedocs itself https://docs.readthedocs.io/en/stable/tutorial/)
- Getting Started
- Bedrock
- Mozorg Mode
- Pocket Mode

Shared stuff goes under bedrock and mode-specific stuff is easy to find at a glance. Getting started is general intro and all the installation instructions (which you'll only really need once).

I'd also like to move a lot of subsections under the bigger pages into their own pages (Images, Macros, etc.).

I'm using "mode" here for Pocket and Mozorg but we also use it to describe running Dev and [Prod](https://bedrock.readthedocs.io/en/latest/install.html#prod-mode), so it might be nice to have a different word to distinguish those contexts. ie right now if I ask what mode you are running, you don't know if I mean Mozorg/Pocket or Dev/Prod

And I'm very interested in hearing your thoughts on this too! Thought we'd keep an open thread here until next docs day and decide how to move forward then

**Comment by stevejalim** [comment]
Would be good to into the idea expressed clearly at the top of/before Getting Started that Bedrock is a platform and Mozorg and Pocket are modes/ways of using it, because that should help with everything that follows

**Comment by reemhamz** [comment]
@stevejalim I like that idea. It would also help solidify the language of how the project itself is structured.

**Comment by maureenlholland** [comment]
Maybe some lessons to be learned from MDN revamp? https://hacks.mozilla.org/2022/10/revamp-of-mdn-web-docs-contribution-docs/

**Comment by maureenlholland** [comment]
closing as outdated, better to focus on https://github.com/mozmeao/springfield/issues/23

**Comment by janbrasna** [comment]
(I think it's covered enough by #11390 otherwise.)

I find the docs organized well. The longer pages (as setup, env, dev and friends) make sense in the current form, as… they're just a longer read, but necessary, and easy to follow.

_(I would disagree these are for initial setup only, I keep coming back to refer to things quite often personally… as I just don't remember all of that;D 🙈)_

I've noticed a handful of hints mentioning outdated tools or not mentioning newer additions that I'll try to collect and post to the other thread not to lose track, but nothing significant really. (Mostly I had issues with postgres tooling, which might be a little OS-dependent.)

There are also a few recent formatting woes from mass-converting from rST to MD making weird output e.g. for _abbr_, but also some reference links etc. — I have some occurrences noted, I'll try to find a replacement markup for these if we can support them properly without adding a plugin or something.

TL;DR is: I did not find any of the current organization limiting in my learning curve. The experience was wonderful and all made sense, and was very efficient and effective to follow and pick up. _(If anything, it might have been the outdated or unused image helpers mentioned in the other thread too, or some implicit behav thereof being a little bit of a tribal knowledge or only available to infer from source code or previous art. That's my only recollection of a little bit of a friction that I had to learn from trial & error personally.)_

**Comment by stephaniehobson** [comment]
@janbrasna I'm so happy to hear that they are legible to someone doing initial setup :D 

I will also +1 that I use them daily myself.
